pkg/scheduling/requirement.go

```diff
@@ -40,6 +40,20 @@ func NewRequirement(key string, operator v1.NodeSelectorOperator, values ...stri
 	if normalized, ok := v1alpha5.NormalizedLabels[key]; ok {
 		key = normalized
 	}
+
+	// This is a super-common case, so optimize for it an inline everything.
+	if operator == v1.NodeSelectorOpIn {
+		s := make(sets.Set[string], len(values))
+		for _, value := range values {
```

Comment:
Is this tangibly different than calling `Insert`?

Reply:
Yes, a bit faster:
```
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
BenchmarkManual-12    	12869192	       122.4 ns/op
BenchmarkNew-12       	14114200	       154.7 ns/op
BenchmarkInsert-12    	13081554	       136.3 ns/op
PASS
```
```
func BenchmarkManual(b *testing.B) {
	values := make([]string, b.N)
	for i := 0; i < b.N; i++ {
		values[i] = strconv.Itoa(i)
	}

	b.ResetTimer()
	s := make(sets.Set[string], len(values))
	for _, value := range values {
		s[value] = sets.Empty{}
	}
	_ = s
}

func BenchmarkNew(b *testing.B) {
	values := make([]string, b.N)
	for i := 0; i < b.N; i++ {
		values[i] = strconv.Itoa(i)
	}

	b.ResetTimer()
	s := sets.New(values...)
	_ = s
}
func BenchmarkInsert(b *testing.B) {
	values := make([]string, b.N)
	for i := 0; i < b.N; i++ {
		values[i] = strconv.Itoa(i)
	}

	b.ResetTimer()
	s := make(sets.Set[string], len(values))
	s.Insert(values...)
	_ = s
}
```